portfolio/src/main/webapp/script.js

```diff
@@ -26,3 +26,94 @@ function addRandomGreeting() {
   const greetingContainer = document.getElementById('greeting-container');
   greetingContainer.innerText = greeting;
 }
+
+/*
+ In order to access the secret page you have to solve a randomly generated equation.
+ Upon the input of a correct answer a button will appear, linking the user to the page.
+ */
+function secretPage(){
+  //Generates a number between 1 and 200 for both numbers.
+  this.a = Math.floor((Math.random()*200)+100);
+  this.b = Math.floor((Math.random()*200)+100);
+
+
+  //If b is a greater number we flip the values
+  if(this.b > this.a){
+    var hold =this.b;
+    this.b = this.a;
+    this.a = hold;
+  }
+
+  // This section of code includes the possible equations to solve
+  this.div = function(){
```

Comment:
This is an interesting way of doing this. It's good to demonstrate how to save functions to a variable (called closures in Javascript) but in this context it makes the code more complex and harder to read. Instead of saving these operations to function closures, you could directly do perform the operation in the switch statement on line 72

Reply:
Got it.